SSA/Experimental/ASTPrettyPrinter.lean

```diff
@@ -0,0 +1,278 @@
+  | _ => throw ()
+
+/--
+info: [mlir_op_successor_arg|^bb] : BBName
+-/
+#guard_msgs in #check [mlir_op_successor_arg| ^bb]
+
+
+-- EDSL MLIR TYPES
+-- ===============
+
+-- TODO: Hardcoded meta-variable case?
+@[app_unexpander AST.MLIRType.int]
+def unexpandMLIRTypeint : Unexpander
+  | `($_ Signedness.Signless $n:num) =>
+    let xid := mkIdent $ Name.mkSimple  ("i" ++ n.getNat.repr)
+    `([mlir_type|$xid:ident])
+  | _ => throw ()
+
+#check [mlir_type| i32]
+
+
+@[app_unexpander AST.MLIRType.float]
+def unexpandMLIRTypefloat : Unexpander
+  | `($_ $n:num) =>
+    let xid := mkIdent $ Name.mkSimple  ("f" ++ n.getNat.repr)
+    `([mlir_type|$xid:ident])
+  | _ => throw ()
+
+#check [mlir_type| f32]
+
+
+@[app_unexpander AST.MLIRType.index]
+def unexpandMLIRTypefindex : Unexpander
+  | `($_ ) =>
+    let id := mkIdent $ Name.mkSimple  ("index")
+    `([mlir_type|$id:ident])
+
+#check [mlir_type| index]
+
+
+@[app_unexpander AST.MLIRType.undefined]
+def unexpandMLIRTypeundefined : Unexpander
+  | `($_ $x:str) =>
+    `([mlir_type| ! $x:str ])
+  | `($_ $x:ident) =>
+    `([mlir_type| ! $x:ident ])
+  | _ => throw ()
+
+-- unexpander currently has no way of determining between idents and strings because
+-- the macro sends them both to strings
+#check [mlir_type| !shape.value]
+#check [mlir_type| !"lz.int"]
+
+
+@[app_unexpander  AST.MLIRType.tensor1d]
+def unexpandMLIRTypetensor1d : Unexpander
+  | `($_ ) =>
+      `([mlir_type|tensor1d])
+
+#check [mlir_type| tensor1d]
+
+
+@[app_unexpander  AST.MLIRType.tensor2d]
+def unexpandMLIRTypetensor2d : Unexpander
+  | `($_ ) =>
+      `([mlir_type|tensor2d])
+
+#check [mlir_type| tensor2d]
+
+
+-- === VECTOR TYPE ===
+--skipping vector <> for now because it may currently have bugs
+
+
+
+-- EDSL MLIR USER ATTRIBUTES
+-- =========================
+
+
+-- EDSL MLIR BASIC BLOCK OPERANDS
+-- ==============================
+
+
+-- EDSL MLIR BASIC BLOCKS
+-- ======================
+
+
+def stringToMLIRuniform_op(op_name : String): UnexpandM (TSyntax `MLIR.Pretty.uniform_op) :=
+  match op_name with
+    | "llvm.return" => `(MLIR.Pretty.uniform_op|llvm.return)
+    | "llvm.copy"   => `(MLIR.Pretty.uniform_op|llvm.copy)
+    | "llvm.neg"    => `(MLIR.Pretty.uniform_op|llvm.neg)
+    | "llvm.not"    => `(MLIR.Pretty.uniform_op|llvm.not)
+    --| "llvm.add"    => `(MLIR.Pretty.exact_op|llvm.add)
+    | "llvm.and"    => `(MLIR.Pretty.uniform_op|llvm.and)
+    --| "llvm.ashr"   => `(MLIR.Pretty.uniform_op|llvm.ashr)
+    --| "llvm.lshr"   => `(MLIR.Pretty.uniform_op|llvm.lshr)
+    --| "llvm.mul"    => `(MLIR.Pretty.uniform_op|llvm.mul)
+    --| "llvm.or"     => `(MLIR.Pretty.uniform_op|llvm.or)
+    --| "llvm.sdiv"   => `(MLIR.Pretty.uniform_op|llvm.sdiv)
+    --| "llvm.shl"    => `(MLIR.Pretty.uniform_op|llvm.shl)
+    | "llvm.srem"   => `(MLIR.Pretty.uniform_op|llvm.srem)
+    --| "llvm.sub"    => `(MLIR.Pretty.uniform_op|llvm.sub)
+    --| "llvm.udiv"   => `(MLIR.Pretty.uniform_op|llvm.udiv)
```

Comment:
For some reason the commented lines caused errors during compilation, hence they were disabled.